src/main/scala/gov/wicourts/json/formlet/Formlet.scala

```diff
@@ -202,84 +225,90 @@ object Formlet extends FormletSyntax {
     N[_, _]: Bifunctor,
     I,
     V: Monoid,
-    E: Semigroup
+    E: Semigroup,
   ](
-    implicit NN: Applicative[N[E, ?]]
+    implicit NN: Applicative[N[E, ?]],
   ): Applicative[Formlet[M, N, I, V, E, ?]] =
     new Applicative[Formlet[M, N, I, V, E, ?]] {
+
       override def map[A, B](a: Formlet[M, N, I, V, E, A])(f: A => B): Formlet[M, N, I, V, E, B] =
-        a map f
+        a.map(f)
 
-      override def point[A](a: => A): Formlet[M, N, I, V, E, A] =
-        Formlet[M, N, I, V, E, A](_ => Applicative[M].point((NN.point(a), Monoid[V].zero)))
+      override def pure[A](a: A): Formlet[M, N, I, V, E, A] =
+        Formlet[M, N, I, V, E, A](_ => Applicative[M].pure((NN.pure(a), Monoid[V].empty)))
 
       override def ap[A, B](
-        fa: => Formlet[M, N, I, V, E, A]
+        f: Formlet[M, N, I, V, E, A => B],
       )(
-        f: => Formlet[M, N, I, V, E, A => B]
+        fa: Formlet[M, N, I, V, E, A],
       ): Formlet[M, N, I, V, E, B] =
-        fa ap f
+        fa.ap(f)
     }
 
   implicit def formletDisjunctionMonad[
     M[_]: Monad,
     I,
     V: Monoid,
-    E: Semigroup
-  ]: Monad[Formlet[M, \/, I, V, E, ?]] =
-    new Monad[Formlet[M, \/, I, V, E, ?]] {
-      override def map[A, B](a: Formlet[M, \/, I, V, E, A])(f: A => B): Formlet[M, \/, I, V, E, B] =
+    E: Semigroup,
+  ]: Monad[Formlet[M, Either, I, V, E, ?]] =
+    new Monad[Formlet[M, Either, I, V, E, ?]] with StackSafeMonad[Formlet[M, Either, I, V, E, ?]] {
```

Comment:
Is it stack-safe, though?

Reply:
I think it was. But had enough doubts to rewrite it